hal/mbed_ticker_api.c

```diff
@@ -121,14 +121,35 @@ static void update_present_time(const ticker_data_t *const ticker)
     uint64_t elapsed_ticks = (ticker_time - queue->tick_last_read) & queue->bitmask;
     queue->tick_last_read = ticker_time;
 
-    uint64_t us_x_ticks = elapsed_ticks * 1000000;
-    uint64_t elapsed_us = us_x_ticks / queue->frequency;
-
-    // Update remainder
-    queue->tick_remainder += us_x_ticks - elapsed_us * queue->frequency;
-    if (queue->tick_remainder >= queue->frequency) {
-        elapsed_us += 1;
-        queue->tick_remainder -= queue->frequency;
+    uint64_t elapsed_us;
+    if (1000000 == queue->frequency) {
```

Comment:
Does it actually change much? Or do we just make to code less readable? Maybe at least we could move that to separate utility functions.

Reply:
This optimization is extremely important. Without this optimization some boards, such as the nrf51 are too slow to pass some of the timing tests.